plugins/woocommerce/src/StoreApi/Utilities/CheckoutTrait.php

```diff
@@ -85,7 +85,13 @@ private function process_payment( \WP_REST_Request $request, PaymentResult $paym
 				throw new RouteException( 'woocommerce_rest_checkout_invalid_payment_result', __( 'Invalid payment result received from payment method.', 'woocommerce' ), 500 );
 			}
 		} catch ( \Exception $e ) {
-			throw new RouteException( 'woocommerce_rest_checkout_process_payment_error', esc_html( $e->getMessage() ), 400 );
+			if ( $e->getPrevious() ) {
```

Comment:
`$additional_data` is defined inside the branch. Line 94 will error out if the exception doesn't go into the condition, as the variable will be undefined.

Reply:
Thank you for catching that! Fixed here [86d1eef](https://github.com/woocommerce/woocommerce/pull/47489/commits/86d1eefb49bccb431c1becb4c52f39f7cf969970)